packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,314 @@
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { ListboxClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    ListboxHandledProps,
+    ListboxProps,
+    ListboxUnhandledProps,
+} from "./listbox.props";
+import * as React from "react";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { get, inRange, invert } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+
+export interface ListboxState {
+    /**
+     * The index of the focusable child
+     */
+    focusIndex: number;
+}
+
+export interface TypeAheadDataItem {
+    index: number;
+    compareString: string;
+}
+
+class Listbox extends Foundation<
+    ListboxHandledProps,
+    ListboxUnhandledProps,
+    ListboxState
+> {
+    public static displayName: string = "Listbox";
+
+    protected handledProps: HandledProps<ListboxHandledProps> = {
+        children: void 0,
+        managedClasses: void 0,
+        typeAheadPropName: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private typeAheadString: string = "";
+    private typeAheadTimer: any;
+    private typeAheadData: TypeAheadDataItem[] = [];
+
+    constructor(props: ListboxProps) {
+        super(props);
+
+        this.state = {
+            focusIndex: -1,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                role="listbox"
+                className={this.generateClassNames()}
+                onKeyDown={this.handleMenuKeyDown}
+            >
+                {this.renderChildren()}
+            </div>
```

Comment:
what was the reasoning behind this being a `div` and not a `ul`, with every child being an `li` with the `role="option"` applied?

Reply:
Same reasoning as for context-menu.  ie.  ul's only accept li elements as children, we are overriding role and other aria props so it is not necessary for accessiblity.

Reply:
Yes, but you can use the li yourself and wrap the given children, you do not need to rely on the user to pass children that are li's

Reply:
What benefit does this give?  Should we not support any arbitrary non-focusable elements as children?